src/components/forms/SubmissionView.js

```diff
@@ -630,33 +665,46 @@ export default class SubmissionView extends React.PureComponent{
             const linksCopy = _.clone(keyLinks);
             const roundTwoCopy = roundTwoKeys.slice();
             const hierarchy = _.clone(keyHierarchy);
-            let dummyHierarchy = object.deepClone(hierarchy);
-            let hierKey = key;
+            const dummyHierarchy = object.deepClone(hierarchy);
+
+            let keyToRemoveIdx = parseInt(keyToRemove) ? keyToRemove : null;
+            const keyToRemoveAtId = !parseInt(keyToRemove) ? keyToRemove : null;
```

Comment:
Not sure if properly checking what want it to -
I think more likely we want `!isNaN(keyToRemove)` (?)
isNaN returns true if not a number or false if it is a number, regardless of type (string or number).
ParseInt always return an integer, however that integer might be "NaN". ~~I believe "NaN" evaluates to truthy so `parseInt(keyToRemove)` would transitively always be `true`.~~

`NaN` evaluates to "falsy" so we're ok, however parseInt("0") would return 0 which is also falsy. So something to consider.

```javascript
>> parseInt("abcd")
NaN
>> !!(parseInt("abcd"))
false
>> !!(parseInt("123"))
true
>> !!(parseInt("0"))
false
>> isNaN("1")
false
>> isNaN("a")
true
```